cmd/gapit/create_graph_visualization.go

```diff
@@ -0,0 +1,74 @@
+// Copyright (C) 2018 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"flag"
+	"github.com/google/gapid/core/app"
+	"github.com/google/gapid/core/log"
+	"io/ioutil"
+	"os"
+)
+
+type createGraphVisualizationVerb struct{ CreateGraphVisualizationFlags }
+
+func init() {
+	verb := &createGraphVisualizationVerb{}
+	app.AddVerb(&app.Verb{
+		Name:      "create_graph_visualization",
+		ShortHelp: "Create graph visualization file from capture",
+		Action:    verb,
+	})
+}
+
+func (verb *createGraphVisualizationVerb) Run(ctx context.Context, flags flag.FlagSet) error {
+	if flags.NArg() != 1 {
+		app.Usage(ctx, "Exactly one gfx trace file expected, got %d", flags.NArg())
+		return nil
+	}
+
+	client, capture, err := getGapisAndLoadCapture(ctx, verb.Gapis, GapirFlags{}, flags.Arg(0), CaptureFileFlags{})
+	if err != nil {
+		return err
+	}
+	defer client.Close()
+
+	log.I(ctx, "Creating graph visualization file from capture id: %s", capture.ID)
+
+	graphVisualization, err := client.GetGraphVisualization(ctx, capture)
+	if err != nil {
+		return log.Errf(ctx, err, "ExportCapture(%v)", capture)
+	}
+
+	filePath := verb.Out
+	if filePath == "" {
+		filePath = "graph_visualization.dot"
+	}
+
+	_, err = os.Stat(filePath)
+	if os.IsNotExist(err) {
```

Comment:
`os.Create` is safe to call even if the file already exists. I think the file will be replaced if it already exists.